Include/nosSettingsSubsystem/nosSettingsSubsystem.h

```diff
@@ -6,51 +6,56 @@
 #define NOS_SETTINGS_SUBSYSTEM_H_INCLUDED
 #include "Nodos/Types.h"
 
-typedef enum nosSettingsFileDirectory {
-	NOS_SETTINGS_FILE_DIRECTORY_LOCAL, // Module's root folder
-	NOS_SETTINGS_FILE_DIRECTORY_WORKSPACE, // Engine's config folder
-	NOS_SETTINGS_FILE_DIRECTORY_GLOBAL	// AppData folder
-} nosSettingsFileDirectory;
-
-typedef nosResult(*nosPfnSettingsItemUpdate)(const char* entryName, nosBuffer itemValue);
+// NOS_RESULT_SUCCESS: Item value will be serialized to the WriteDirectories
+// NOS_RESULT_FAIL: Default item value will be serialized to the WriteDirectories
+typedef nosResult(*nosPfnSettingsEntryUpdate)(nosName entryName, nosBuffer itemValue);
+typedef uint64_t nosSettingsEntryId;
 
 #if NOS_HAS_CPLUSPLUS_20
-#include "EditorEvents_generated.h"
-typedef nos::sys::settings::editor::SettingsEditorItem nosSettingsEditorItem;
+typedef nos::fb::Visualizer* nosSettingsEditorVisualizer;
 #else
-typedef void* nosSettingsEditorItem;
+typedef void* nosSettingsEditorVisualizer;
 #endif
 
+typedef enum nosSettingsFileDirectory {
+	NOS_SETTINGS_FILE_DIRECTORY_LOCAL = 1 << 0, // Module's root folder
+	NOS_SETTINGS_FILE_DIRECTORY_WORKSPACE = 1 << 1, // Engine's config folder
+	NOS_SETTINGS_FILE_DIRECTORY_GLOBAL = 1 << 2	// AppData folder
+} nosSettingsFileDirectory;
+
+typedef uint32_t nosSettingsFileDirectoryFlag;
+
 typedef struct nosSettingsEntryParams {
 	nosName TypeName;
-	nosBuffer Buffer; // Don't forget to free this buffer
-	const char* EntryName; // If NULL, "default" will be used
-	nosSettingsFileDirectory Directory;
+	// Default value of the buffer
+	// If nosPfnSettingsEntryUpdate callback returns NOS_SETTINGS_ENTRY_NON_COMPATIBLE, this will be saved
+	// Don't forget to free this buffer
+	const nosBuffer* Buffer;
+	nosName EntryName; // If NULL, "default" will be used
+	nosSettingsFileDirectoryFlag WriteDirectories; // Directory flags to determine where this entry will be stored
+	nosPfnSettingsEntryUpdate UpdateCallback; // Callback to update the entry value
+	bool IsEditableFromEditor; // If true, this entry will be editable from editor
```

Comment:
We can delete this I think. If read-only is needed for some setting we would add it later.